core/src/test/java/org/elasticsearch/index/query/FilteredQueryBuilderTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.ConstantScoreQuery;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.lucene.search.Queries;
+
+import java.io.IOException;
+
+@SuppressWarnings("deprecation")
+public class FilteredQueryBuilderTest extends BaseQueryTestCase<FilteredQueryBuilder> {
+
+    @SuppressWarnings("rawtypes")
+    @Override
+    protected FilteredQueryBuilder createTestQueryBuilder() {
+        QueryBuilder queryBuilder = RandomQueryBuilder.create(random());
```

Comment:
So currently `RandomQueryBuilder.create(random())` doesn't return `null`, but I see you already plan for this in `testToQuery`. Maybe don't throw the switch in the random query generator just yet because I'm thinking about changes there (see comment above), but maybe randomly add `null` just here in createTestQueryBuilder so it won't blow up the other tests.